arrow-libs/optics/arrow-optics-ksp-plugin/src/main/kotlin/arrow/optics/plugin/internals/utils.kt

```diff
@@ -6,3 +6,103 @@ package arrow.optics.plugin.internals
  */
 fun String.plusIfNotBlank(prefix: String = "", postfix: String = "") =
   if (isNotBlank()) "$prefix${this}$postfix" else this
+
+/**
+ * Sanitizes each delimited section if it matches with Kotlin reserved keywords.
+ */
+fun String.sanitizeDelimited(delimiter: String = ".", separator: String = delimiter) =
+  if (isNotBlank() && contains(delimiter)) {
+    splitToSequence(delimiter).joinToString(separator) { if (kotlinKeywords.contains(it)) "`$it`" else it }
+  } else this
```

Comment:
I don't think you need to consider empty and not containing delimiter as special cases, `splitToSequence` + `joinToString` should do the right thing